noir-projects/aztec-nr/aztec/src/context/private_context.nr

```diff
@@ -249,28 +249,49 @@ impl PrivateContext {
     ) -> [Field; RETURN_VALUES_LENGTH] {
         let args_hash = hash_args(args);
         assert(args_hash == arguments::pack_arguments(args));
-        self.call_private_function_with_packed_args(contract_address, function_selector, args_hash)
+        self.call_private_function_with_packed_args(contract_address, function_selector, args_hash, false)
+    }
+
+    pub fn call_private_function_static<ARGS_COUNT>(
+        &mut self,
+        contract_address: AztecAddress,
+        function_selector: FunctionSelector,
+        args: [Field; ARGS_COUNT]
+    ) -> [Field; RETURN_VALUES_LENGTH] {
+        let args_hash = hash_args(args);
+        assert(args_hash == arguments::pack_arguments(args));
+        self.call_private_function_with_packed_args(contract_address, function_selector, args_hash, true)
     }
 
     pub fn call_private_function_no_args(
         &mut self,
         contract_address: AztecAddress,
         function_selector: FunctionSelector
     ) -> [Field; RETURN_VALUES_LENGTH] {
-        self.call_private_function_with_packed_args(contract_address, function_selector, 0)
+        self.call_private_function_with_packed_args(contract_address, function_selector, 0, false)
+    }
+
+    pub fn call_private_function_no_args_static(
+        &mut self,
+        contract_address: AztecAddress,
+        function_selector: FunctionSelector,
+    ) -> [Field; RETURN_VALUES_LENGTH] {
+        self.call_private_function_with_packed_args(contract_address, function_selector, 0, true)
     }
 
     pub fn call_private_function_with_packed_args(
         &mut self,
         contract_address: AztecAddress,
         function_selector: FunctionSelector,
-        args_hash: Field
+        args_hash: Field,
+        is_static_call: bool,
```

Comment:
If other APIs have a standard version and a `_static` version, then this one should probably be the same. I.e., move the shared logic to an internal function and expose both `call_private_function_with_packed_args` and `call_private_function_with_packed_args_static`.

Reply:
I'm also wondering if we should name it `static_call_private_function` and `delegate_call_private_function` 🤔

Reply:
I don't think this one is used externally, is is simply a thin layer of abstraction but only for us. I can still separate them, but I don't think other devs should be using it anyways.

Reply:
Yeah you are right this one is not used externally and probably won't. In that case, maybe remove the `pub` so it's clear it's not publicly facing!